Documentation/convert ms (#72)
qbicsoftware/etl-scripts: drop-boxes/register-convert-ms-vendor-format/README.md

# Mass Spectrometry Dropbox

## Expected data structure
In every use case, the data structure needs to contain a top folder around the respective data in order to accommodate metadata files.

The sample code found in the top folder can be of type `Q_TEST_SAMPLE` or `Q_MS_RUN`. In the former case, a new sample of type `Q_MS_RUN` is created and attached as child to the test sample.

**Valid folder/file types**:
- Thermo Fisher Raw file format
- Waters Raw folder
- Bruker .d folder

**Incoming structure overview for standard case without additional metadata file:**
```

> QABCD102A5_20201229145526_20201014_CO_0976StSi_R05_.raw

Reviewer: why does the top folder name has a file ending?

Author: good point, this is confusing. it happens because the datahandler automatically creates a folder from the file (or folder) name of the incoming data and forwards it to this and other dropboxes.
this script then expects that the main data has the same name as the top folder, so this information is important.

Reviewer: I see, thanks for the explanation

|-- QABCD102A5_20201229145526_20201014_CO_0976StSi_R05_.raw
|-- QABCD102A5_20201229145526_20201014_CO_0976StSi_R05_.raw.sha256sum
```
In this case, existing mass spectrometry metadata is expected to be already stored and the dataset will be attached.

> **Incoming structure overview for case for Immunomics data with metadata file:**

Reviewer: ```suggestion
**Incoming structure overview for the use case of Immunomics data with metadata file:**
```

```
QABCD090B7
|-- QABCD090B7
|   |-- file1.raw
|   |-- file2.raw
|   |-- file3.raw
|   `-- metadata.tsv
|-- QABCD090B7.sha256sum

> `-- source_dropbox.txt

Reviewer: what is about the metadata json as stated [here](https://github.com/qbicsoftware/etl-scripts/blob/b65dacc8abfa946198f6367e7fb2bcf1f4925a92/drop-boxes/register-convert-ms-vendor-format/etl_msconvert.py#L292) in the comment of one method?

Author: this is unused and seems to be from a really early stage of the script. nicely spotted!

Author: what shall we do about it?

Reviewer: hm.. if its not used I would maybe also remove it out of the code at one point?
I thought json schemas are our golden way/we aim for them?

Author: it's really, really old...I will open an issue for it :)

Reviewer: sorry for being so nitty 🙈  but I think even if it's old it helps to clean up the code and not lose track of the issue :)

Author: no no, I agree, I mean the golden way of json was not a thing back then, hence we went with something the labs wanted to use instead :)

```
The source_dropbox.txt currently has to indicate the source as one of the Immunomics data sources.

> metadata.tsv columns for the Immunomics case, tab-separated with one example row:

Reviewer: ```suggestion
The `metadata.tsv` columns for the Immunomics case are tab-separated:
```

```
Filename	Q_MS_DEVICE	Q_MEASUREMENT_FINISH_DATE	Q_EXTRACT_SHARE	Q_ADDITIONAL_INFO	Q_MS_LCMS_METHODS	technical_replicate	workflow_type
file1.raw	THERMO_QEXACTIVE	171010	10		QEX_TOP07_470MIN	DDA_Rep1	DDA
```

Filename - one of the (e.g. raw) file names found in the incoming structure
Q_MS_DEVICE - openBIS code from the vocabulary of Mass Spectrometry devices
Q_MEASUREMENT_FINISH_DATE - Date in YYMMDD format (ISO 8601:2000)
Q_EXTRACT_SHARE - the extract share
Q_ADDITIONAL_INFO - any optional comments
Q_MS_LCMS_METHODS - openBIS code from the vocabulary of LCMS methods
technical_replicate - free text to denote replicates
workflow_type - DDA or DIA

**Resulting data model in openBIS**  

> ...Q_TEST_SAMPLE (-> Q_MHC_LIGAND_EXTRACT (Immunomics case)) -> Q_MS_RUN per data file --> 2 DataSets per data file, one for raw data, one converted to mzML

Reviewer: maybe put this on top of the document to have the same structure for all documentations as in the main [README](https://github.com/qbicsoftware/etl-scripts/blob/master/README.md) where this documentation should also be put.

Author: will do
